### PR TITLE
Release docs for 2.5

### PR DESCRIPTION
**Pull request by richdougherty**
I edited the existing highlights and migration docs and added some new text.

There's a lot of content we need to write so I put placeholder `TODO` entries in the docs for us to fill in later.

This PR should be merged even though it's incomplete so that people can start working on the TODO entries.

### CODE REVIEW

File: documentation/manual/releases/release25/Highlights25.md [hunk 1]
```diff
@@ -3,15 +3,70 @@
...
+
+- body parsers, filters, WS, WebSockets
+
+- explain performance improvements for no-body request and chunked responses
+
+- link to information in migration docs
+
+## Java API updated to use Java 8 classes
+
+When Play 2.0 was released in 2013 Java had little support for Play's style of asynchronous functional programming. There were no lambdas, futures only had a blocking interface and common functional classes didn't exist. Play provided its own classes to fill the gap.
```
Comment (mkurz) [inline documentation/manual/releases/release25/Highlights25.md]: Play 2.0 was released in 2012 :wink:

Comment (richdougherty) [inline documentation/manual/releases/release25/Highlights25.md]: OK, fixed!

File: documentation/manual/releases/release25/Migration25.md [hunk 2]
```diff
@@ -3,96 +3,204 @@
...
+
+
+## Renamed WS AsyncHttpClient classes
+
+The [AsyncHttpClient](https://github.com/AsyncHttpClient/async-http-client) is the technology underneath Play's WS client. Many of Play's WS classes used the outdated name "Ning" to refer to the AsyncHttpClient library. To properly reflect the AsyncHttpClient project name Play 2.5 renames or deprecates classes using the name "Ning".
+
+### How to migrate
+
+**TODO: Check the list of classes and config settings that have changed and document them here. Check which bits are deprecated. I think config using the old name will give a warning at runtime.** More info in the PR: https://github.com/playframework/playframework/pull/5224. **Some text we might want to use:** "package `play.api.libs.ws.ning` was renamed into `play.api.libs.ws.ahc` and `Ning*` classes were renamed into `Ahc*`."
+## Deprecated `GlobalSettiongs`
```
Comment (gmethvin) [inline documentation/manual/releases/release25/Migration25.md]: `GlobalSettings`

**Comment by gmethvin** [comment]
Looks great so far.

File: documentation/manual/releases/release25/Migration25.md (hunk 2 above)
Comment (richdougherty) [inline documentation/manual/releases/release25/Migration25.md]: Fixed.

**Pull request merged**